### PR TITLE
Feature gui

### LINKED ISSUE #16: [Client] User Interface
## Description
Implement a user interface for the web client.

### PR DESCRIPTION
**Pull request by dtoniuc**
#### What does this PR do?
This PR adds a graphical user interface to the system. The first commits separate the network logic from the ui logic. Then the gui classes are added. The last commits fix the build scripts for the entire system

#### Where should the reviewer start?
The reviews should start with the CryptoClient class and should check the new method signatures. Than the ConsoleLauncher class should be checked to ensure that the console user interface still works properly. Finally the reviews should analyze the gui classes.

#### How should this be manually tested?
1. All methods should be tested from the console interface (use script ./start.sh to start the client)
2. All methods should be tested from graphical interface (use script ./startGui.sh to start the client)

#### Any background context you want to provide?
The graphical user interface uses JavaFX with fxml files

#### What are the relevant tickets?
Closes #16 

#### Screenshots
screenshot from 2017-10-19 22-25-25
screenshot from 2017-10-19 22-25-15

### CODE REVIEW

File: JEasyCrypto/build.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
+mkdir bin
```
Comment (a-rmz) [inline JEasyCrypto/build.sh]: For *NIX systems, you can add the `-p` flag so it doesn't throw any error if the directory already exists.
I.e:
```bash
mkdir -p bin
```

File: JEasyCryptoClient/build.sh [hunk 2]
```diff
@@ -1 +1,5 @@
-javac src/*.java -classpath "../json-simple-1.1.1.jar:." -d bin
+mkdir bin
```
Comment (a-rmz) [inline JEasyCryptoClient/build.sh]: Same with this `mkdir`.

File: JEasyCryptoClient/startGui.cmd [hunk 3]
```diff
@@ -0,0 +1,2 @@
+java -cp ..\json-simple-1.1.1.jar;bin;. gui.AppLauncher
+
```
Comment (a-rmz) [inline JEasyCryptoClient/startGui.cmd]: No need for this line

File: JEasyCryptoClient/startGui.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+java -cp ../json-simple-1.1.1.jar:bin: gui.AppLauncher
+
```
Comment (a-rmz) [inline JEasyCryptoClient/startGui.sh]: Same here.

File: JEasyCryptoConsole/build.sh [hunk 5]
```diff
@@ -1 +1,2 @@
+mkdir bin
```
Comment (a-rmz) [inline JEasyCryptoConsole/build.sh]: `mkdir -p bin`

File: JEasyCryptoServer/build.sh [hunk 6]
```diff
@@ -1 +1,2 @@
+mkdir bin
```
Comment (a-rmz) [inline JEasyCryptoServer/build.sh]: `mkdir -p bin`

File: JEasyCryptoClient/startGui.cmd [hunk 7]
```diff
@@ -0,0 +1,2 @@
+java -cp ..\json-simple-1.1.1.jar;bin;. gui.AppLauncher
```
Comment (a-rmz) [inline JEasyCryptoClient/startGui.cmd]: As for the name of the file, I'd suggest `start-gui.sh`, with dash, not camel case.

**Comment by dtoniuc** [comment]
Changes done

**Comment by a-rmz** [comment]
Amazing job!!! Thanks! Very well done. :D 👍

**Pull request merged**